docs/DesignDocs/ExtendedEnclaveInitializationData_V2.md

```diff
@@ -0,0 +1,323 @@
+the end of the committed enclave pages, both under the potentially malicious
+loader's control, complicate the EEID implementation and require more security
+design/code review and validation. For usages that do not need to support
+enclave load time selection of heap/stack size and #TCS, an EEID enclave with
+enclave heap/stack size and #TCS specified at enclave signing time is more
+attractive.
+
+This update to the design aims to achieve the following goals:
+
+- Unify the APIs related to manage and access the additional code/data allowed
+  to be loaded into the enclave, between the EEID based solution and HW
+  CONFIGID/CONFIGSVN based solution.
+- Add support for EEID enclave with heap/stack size and #TCS specified at
+  enclave signing time.
+- Simplify EEID design/implementation, especially for EEID enclave with with
+  heap/stack size and #TCS specified at enclave signing time.
+
+Compatibility with KSS based solution
+-------------------------------------
+
+To expose the same programing interface as the future KSS based solution, the
+variable length configuration data is not included in the EEID page anymore.
+Instead, config_id and config_svn are included in the EEID page, and the size of
+EEID data structure is limited to 4KB to fit within one 4KB page.
+
+The `oe_eeid_t` definition is changed from:
+
+```C
+#ifdef EXPERIMENTAL_EEID
+typedef struct oe_eeid_t_
+{
+    uint32_t version;        /* version number of the structure */
+    uint32_t hash_state[10]; /* internal state of the hash computation at the
+                              end of the enclave base image */
+    uint64_t signature_size; /* size of signature */
+    oe_enclave_size_settings_t size_settings; /* heap, stack and thread
+                                                 configuration for this
+                                                 instance */
+    uint64_t vaddr;          /* location of the added data pages in enclave
+                                memory; EEID follows immediately thereafter */
+    uint64_t entry_point;    /* entry point of the image, for SGX, matches
+                                TCS.OENTRY */
+    uint64_t data_size;      /* size of application EEID */
+    uint8_t data[];          /* Buffer holding EEID data followed by the
+                                signature */
+} oe_eeid_t;
+#endif
+```
+
+to:
+
+```C
+#ifdef EXPERIMENTAL_EEID
+typedef struct oe_eeid_t_
+{
+    uint32_t version;        /* version number of the structure */
+    uint32_t hash_state[10]; /* internal state of the hash computation at the
+                                end of the enclave base image */
+    uint8_t config_id[32];   /* identity of the code/data allowed to be
+                                loaded into the enclave after enclave init */
+    unit16_t config_svn;     /* minimum SVN of the code/data allowed to be
+                                loaded into the enclave after enclave init */
```

Comment:
config_svn appears in sgx_key_request_t.  Does it really apply to all TEEs?  If so, should expand the "SVN" acronym in the comment.

Reply:
With the SGX KSS feature, EGETKEY keys can be bound to Config_id and config_svn, based on enclave code's selection (sgx_key_request_t). EEID's config_id/config_svn will not provide the same capability. The latest update to the design doc defines eeid.config_svn as "Optional subsidiary info about the identity of the additional content, for example, a version number".